eth/core/processor.go

```diff
@@ -139,20 +141,17 @@ func (sp *StateProcessor) Prepare(evm *vm.GethEVM, header *types.Header) {
 func (sp *StateProcessor) ProcessTransaction(
 	ctx context.Context, tx *types.Transaction,
 ) (*ExecutionResult, error) {
-	var (
-		// We set the gasUsed to the amount of gas so far used in the block.
-		gasUsed = sp.gp.BlockGasConsumed()
-		// We set the gasPool = gasLimit - gasUsed.
-		gasPool = GasPool(sp.header.GasLimit - gasUsed)
-	)
+	// We set the gasPool = gasLimit - gasUsed.
+	gasPool := new(GasPool).AddGas(sp.header.GasLimit - sp.gp.BlockGasConsumed())
 
 	// Set the transaction context in the state database.
 	// This clears the logs and sets the transaction info.
 	sp.statedb.SetTxContext(tx.Hash(), len(sp.txs))
 
 	// Inshallah we will be able to apply the transaction.
 	receipt, result, err := ApplyTransactionWithEVMWithResult(
-		sp.evm, sp.cp.ChainConfig(), nil, &gasPool, sp.statedb, sp.header, tx, &gasUsed,
+		sp.evm, sp.cp.ChainConfig(), gasPool, sp.statedb, sp.header.BaseFee,
+		sp.header.Number, sp.sealhash, tx, &sp.header.GasUsed,
```

Comment:
way more efficient to precompute the block hash (doing the full rlpHash algo) once per block rather than for every tx

Reply:
I wouldn't recommend having the state machine modify the unit64 on the header, that is some hood shit.

Reply:
thats how geth does it lol

Reply:
Rly lol link??

Reply:
https://github.com/berachain/polaris-geth/blob/stateful-v1.11.4/miner/worker.go#L868

Reply:
So hood. 

Do we wanna just leave the state processor as is and start building a Miner struct on the side?

Reply:
also https://github.com/flashbots/builder/blob/main/internal/ethapi/api.go#L2309

Reply:
its weird but thats why it takes the pointer to integer

Reply:
I think we can leave this as is and merge if we want to. the miner/worker stuff would require way more upheaval so again smth thats prob best for later